src/commonMain/kotlin/uk/gov/hmrc/helptosavecalculator/Calculator.kt

```diff
@@ -15,26 +15,45 @@
  */
 package uk.gov.hmrc.helptosavecalculator
 
+import com.soywiz.klock.DateTime
+import kotlin.jvm.JvmOverloads
 import uk.gov.hmrc.helptosavecalculator.config.HtSSchemeConfig
+import uk.gov.hmrc.helptosavecalculator.exceptions.InvalidAccountStartDate
+import uk.gov.hmrc.helptosavecalculator.exceptions.InvalidCurrentBalance
+import uk.gov.hmrc.helptosavecalculator.exceptions.InvalidCurrentBonusAmount
 import uk.gov.hmrc.helptosavecalculator.exceptions.InvalidRegularPaymentException
 import uk.gov.hmrc.helptosavecalculator.models.CalculatorResponse
 import uk.gov.hmrc.helptosavecalculator.models.MonthlyBreakdown
+import uk.gov.hmrc.helptosavecalculator.utils.monthsSince
 import uk.gov.hmrc.helptosavecalculator.validation.RegularPaymentValidators
-// @JvmOverloads
-class Calculator constructor(
-    private val regularPayment: Int
-): HtSSchemeConfig() {
+
+class Calculator @JvmOverloads constructor(
```

Comment:
What do you think about moving these parameters to the `run` function, as this object doesn't really need to hold these as state?